core/lib/zksync_core/src/api_server/web3/metrics.rs

```diff
@@ -195,3 +195,61 @@ pub(super) struct PubSubMetrics {
 
 #[vise::register]
 pub(super) static PUB_SUB_METRICS: vise::Global<PubSubMetrics> = vise::Global::new();
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "filter_type", rename_all = "snake_case")]
+pub(super) enum FilterType {
+    Events,
+    Blocks,
+    PendingTransactions,
+}
+
+impl From<TypedFilter> for FilterType {
+    fn from(value: TypedFilter) -> Self {
+        match value {
+            TypedFilter::Events(_, _) => FilterType::Events,
+            TypedFilter::Blocks(_) => FilterType::Blocks,
+            TypedFilter::PendingTransactions(_) => FilterType::PendingTransactions,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct InstalledFilter {
+    pub filter: TypedFilter,
+    guard: Box<GaugeGuard>,
+    start: Instant,
+    // FIXME: Change the type
+    pub last_request: u64,
+}
+
+impl InstalledFilter {
+    pub fn new(filter: TypedFilter, guard: GaugeGuard) -> Self {
+        Self {
+            filter,
+            guard: Box::new(guard),
+            start: Instant::now(),
+            last_request: chrono::Utc::now().naive_utc().timestamp() as u64,
+        }
+    }
+}
+
+impl Drop for InstalledFilter {
+    fn drop(&mut self) {
+        FILTER_METRICS.filter_lifetime[&FilterType::from(self.filter.clone())]
+            .observe(self.start.elapsed());
+    }
+}
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "api_filter")]
+pub(super) struct FilterMetrics {
+    pub metrics_count: Family<FilterType, Gauge>,
+    #[metrics(buckets = Buckets::LATENCIES)]
```

Comment:
For new metrics, you should specify `unit` if possible (e.g., `unit = Unit::Seconds` here and for `filter_lifetime`). The old metrics don't have `unit` set because this would change their names (by convention, unit is added as a suffix to the metric name, e.g. `api_filter_request_frequency_seconds`), so it would break existing Grafana dashboards.